examples/hf_serialization.py

```diff
@@ -226,71 +303,86 @@ def df_main(args: argparse.Namespace) -> None:
     logger.info("GPU: " + utils.get_gpu_name())
     logger.info("PYTHON USED RAM: " + utils.get_mem_usage())
 
-    serialize_model(
-        pipeline.text_encoder.eval(),
-        pipeline.text_encoder.config,
-        output_prefix,
-        "encoder",
-    )
-    serialize_model(pipeline.vae.eval(), None, output_prefix, "vae")
-    serialize_model(pipeline.unet.eval(), None, output_prefix, "unet")
+    for prefix, component in (
+        ("encoder", pipeline.text_encoder),
+        ("vae", pipeline.vae),
+        ("unet", pipeline.unet),
+    ):
+        component.eval()
+        config = component.config
+        if isinstance(config, dict):
+            config = {k: v for k, v in config.items() if k != "_name_or_path"}
+        serialize_model(
+            component, config, output_prefix, prefix, force=args.force
+        )
 
-    pipeline.tokenizer.save_pretrained(output_prefix)
-    pipeline.scheduler.save_pretrained(output_prefix)
+    serialize_pretrained(
+        pipeline.tokenizer, output_prefix, "tokenizer", force=args.force
+    )
+    serialize_pretrained(
+        pipeline.scheduler, output_prefix, "scheduler", force=args.force
+    )
```

Comment:
So the purpose of `serialize_pretrained` is specifically to support saving artifacts like SD's `scheduler` and `tokenizer`? I assume this is cleaner as they're not configs nor modules.

Reply:
Yes, pretty much. The scheduler can actually be saved as a single JSON file (around 300 to 400 bytes), so it could have its own special code to avoid a zip file and temporary directory, but it seemed unnecessary at this time.
In summary:
- `text_encoder` / `vae` / `unet` — tensors & `config.json`
  - We use `tensorizer` to save these
  - Potentially very, very large, so the optimizations in `tensorizer` are important
- `scheduler` — `config.json` only
  - Supports `.save_pretrained(dir)` and `.from_pretrained(dir)`
  - Roughly around 400 bytes, so downloader optimizations aren't a huge deal
- `tokenizer` — entire directory of files
  - Supports `.save_pretrained(dir)` and `.from_pretrained(dir)`
  - Roughly around 1.5 MB for SD 1.5, and compresses well, so zips work nicely
  - For models that use "[fast tokenizers](https://huggingface.co/learn/nlp-course/chapter6/3)," these can be saved as a single `tokenizer.json` file instead, but not all models support fast tokenizers